aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1499,48 +1496,60 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cpu(const Tensor& self, bool pi
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ triangular_solve ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-template<typename scalar_t>
-static void apply_triangular_solve(Tensor& b, Tensor& A, Tensor& infos, bool upper, bool transpose, bool unitriangular) {
-#ifndef USE_LAPACK
-  AT_ERROR("triangular_solve: LAPACK library not found in compilation");
-#else
-  char uplo = upper ? 'U' : 'L';
-  char trans = transpose ? 'T' : 'N';
-  char diag = unitriangular ? 'U' : 'N';
+DEFINE_DISPATCH(triangular_solve_stub);
+
+// Solves 'input' @ 'result' = 'other'
```

Comment:
This comment's already pretty good but elaborate on the other arguments (clone_input, infos, upper, transpose, unitriangular) too